solutions/src/main/java/ru/sberbank/school/task13/cacheproxy/CacheProxy.java

```diff
@@ -0,0 +1,167 @@
+package ru.sberbank.school.task13.cacheproxy;
+
+import lombok.NonNull;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.lang.reflect.*;
+import java.util.*;
+
+/*
+Реализовать класс, порождающий кеширующий прокси — **CacheProxy** с методом **cache(Object service)**,
+который принимает ссылку на сервис и возвращает кешированную версию этого сервиса*.
+CacheProxy должен тоже принимать в конструкторе некоторые настройки,
+например, рутовую папку в которой хранить файлы, дефолтные настройки кеша и тд.
+Логика по кешированию должна навешиваться с помощью **DynamicProxy**.
+*/
+public class CacheProxy<T> {
+    private String pathRoot;
+
+    public CacheProxy(@NonNull String pathRoot) {
+        if (!new File(pathRoot).isDirectory()) {
+            throw new IllegalArgumentException("Path is not a directory");
+        }
+        this.pathRoot = pathRoot;
+    }
+
+    public T cache(@NonNull T service) {
+        ClassLoader classLoader = ClassLoader.getSystemClassLoader();
+        Class<?>[] classes = service.getClass().getInterfaces();
+        DynamicInvocationHandler dynamicInvocationHandler = new DynamicInvocationHandler(service, pathRoot);
+        return (T) Proxy.newProxyInstance(classLoader, classes, dynamicInvocationHandler);
+    }
+
+    private static class DynamicInvocationHandler implements InvocationHandler {
+        private final Object service;
+        private Map<String, Object> cachedResults = new HashMap<>();
+        private String pathRoot;
+
+        private DynamicInvocationHandler(Object service, String pathRoot) {
+            this.service = service;
+            this.pathRoot = pathRoot;
+        }
+
+        @Override
+        public Object invoke(Object proxy, Method method, Object[] args)
+                throws Throwable {
+            Object ourMethodResult = null;
+            System.out.println("Invoked method: " + method.getName());
+            if (!method.isAnnotationPresent(Cache.class)) {
+                ourMethodResult = method.invoke(service, args);
+            } else {
+                Cache annotationParams = method.getDeclaredAnnotation(Cache.class);
+                if (annotationParams.cacheType() == Cache.cacheType.FILE) {
+                    ourMethodResult = toFile(method, args, annotationParams);
+                } else {
+                    ourMethodResult = toJVMMemory(method, args, annotationParams);
+                }
+            }
+
+            return ourMethodResult;
+        }
+
+        private Object toJVMMemory(Method method, Object[] args, Cache annotationParams) {
```

Comment:
Abbreviation in name &#39;toJVMMemory&#39; must contain no more than &#39;2&#39; consecutive capital letters.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a2009278518>*